kimchi/src/circuits/polynomials/range_check/mod.rs

```diff
@@ -0,0 +1,97 @@
+//! Range check gate
+
+///```text
+/// Range check field element structure:
+///
+///    Each field element a is decomposed into three 88-bit limbs a0, a1, a2 s.t. a = a0a1a2 in
+///    little-endian byte order (i.e. a = a2*2^{2b} + a1*2^b + a0)
+///
+///    L is a 12-bit lookup,
+///    C is a 2-bit crumb.
+///
+///         <----6----> <------8------>
+///    a0 = L L L L L L C C C C C C C C
+///    a1 = L L L L L L C C C C C C C C
+///         <--4--> <------------------20----------------->
+///    a2 = L L L L C C C C C C C C C C C C C C C C C C C C
+///
+/// Input structure:
+///
+///   Row*  Contents**
+///     0   a0
+///     1   a1
+///     2   a2
+///     3   a0,a1,a2
+///     4   b0
+///     5   b1
+///     6   b2
+///     7   b0,b1,b2
+///
+///    (*)  Row offsets
+///    (**) Some part of the limb is contained in this row
```

Comment:
This footnote is important to understanding the table above. I would be tempted to rewrite the whole thing as brief prose, e.g.
```
Each of the first 3 gates checks most of a different range-check input. The final gate performs the remaining checks for all 3 inputs.
```

Reply:
👍🏼